test-data-generator/src/main/scala/za/co/absa/spline/testdatagen/NumericValue.scala

```diff
@@ -20,23 +20,34 @@ sealed trait NumericValue {
   def isExpanded: Boolean = this.isInstanceOf[Constant]
 
   def valueOf(): Int = if (isExpanded) this.asInstanceOf[Constant].value else {
-    throw new Exception("Not expanded")
+    throw new NotImplementedError("Not expanded")
   }
+
+  override def toString: String = valueOf().toString
 }
 case class Constant(value: Int) extends NumericValue
 
 case class Variable(start: Int, end: Int, step: Int) extends NumericValue {
   def expand(): Seq[Constant] = (start to end by step).map(i => Constant(i))
+
+  override def toString: String = s"${start}-${end}|${step}"
 }
 
 object NumericValue {
+  private val numberPattern = "([0-9]+)"
+
   def apply(param: String): NumericValue = {
-    val ConstantPattern = "([0-9]+)".r
-    val VariablePattern = "([0-9]+)-([0-9]+)/([0-9]+)".r
+    val ConstantPattern = numberPattern.r
+    val VariablePattern = (numberPattern + "-" + numberPattern + "/" + numberPattern).r
     param match {
-      case ConstantPattern(c) => Constant(c.toInt)
-      case VariablePattern(start: String, end: String, step: String) => Variable(start.toInt, end.toInt, step.toInt)
-      case _ => throw new Exception("Invalid")
+      case ConstantPattern(c) if c.toInt > 0 => Constant(c.toInt)
+      case ConstantPattern(c) => throw new IllegalArgumentException(s"Invalid value ${c}, number should be positive")
+      case VariablePattern(start: String, end: String, step: String)
+        if start.toInt > 0 && end.toInt > 0 && step.toInt > 0 && end > start =>
+        Variable(start.toInt, end.toInt, step.toInt)
+      case VariablePattern(start: String, end: String, step: String) =>
```

Comment:
Unnecessary object destructuring
 ```suggestion
      case _: VariablePattern =>
```